### ISSUE TITLE
NullReferenceException using IColumnMapper 

### ISSUE THREAD

**Issue by MatthewTrout** [jonwagner/Insight.Database#263]
In the example in the wiki on 'Customizing Object Mapping', it is suggested that returning a string such as "__unmapped__", will prevent any other registered mappers from being used, (since non-null) and will simply cause the field to be unbound ('If the field is not found on the given type, the column/parameter is unbound.').  Since this is exactly the behaviour I required, I went ahead and did this, only to find that a query threw a NullReferenceException.

`return Connection.QuerySql<DocumentRecord>("SELECT * FROM " + TableName);`

stack trace (ish):

```
   at Insight.Database.DbSerializationRule.EvaluateRules(ClassPropInfo prop) in c:\projects.net\Insight\Insight.Database\Insight.Database\Serialization\DbSerializationRule.cs:line 220
   at Insight.Database.DbSerializationRule.GetSerializer(IDataReader reader, Int32 column, ClassPropInfo prop) in c:\projects.net\Insight\Insight.Database\Insight.Database\Serialization\DbSerializationRule.cs:line 210
   at Insight.Database.ColumnMapping.MapColumn(Type type, IDataReader reader, Int32 column, IColumnMapper overrideMapping) in c:\projects.net\Insight\Insight.Database\Insight.Database\Mapping\ColumnMapping.cs:line 166
   at Insight.Database.ColumnMapping.<>c__DisplayClass4.<MapColumns>b__3(Int32 i) in c:\projects.net\Insight\Insight.Database\Insight.Database\Mapping\ColumnMapping.cs:line 83
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Insight.Database.ColumnMapping.MapColumns(Type type, IDataReader reader, Int32 startColumn, Nullable`1 columnCount, IColumnMapper overrideMapping) in c:\projects.net\Insight\Insight.Database\Insight.Database\Mapping\ColumnMapping.cs:line 82
   at Insight.Database.CodeGenerator.ClassDeserializerGenerator.MapColumns(Type type, IDataReader reader, Int32 startColumn, Int32 columnCount, IRecordStructure structure, Boolean allowBindChild) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\ClassDeserializerGenerator.cs:line 351
   at Insight.Database.CodeGenerator.ClassDeserializerGenerator.CreateClassDeserializerDynamicMethod(Type type, IDataReader reader, IRecordStructure structure, Int32 startColumn, Int32 columnCount, Boolean createNewObject, Boolean isRootObject, Boolean allowBindChild) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\ClassDeserializerGenerator.cs:line 109
   at Insight.Database.CodeGenerator.ClassDeserializerGenerator.CreateClassDeserializer(Type type, IDataReader reader, IRecordStructure structure, Int32 startColumn, Int32 columnCount, Boolean createNewObject) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\ClassDeserializerGenerator.cs:line 77
   at Insight.Database.CodeGenerator.ClassDeserializerGenerator.CreateDeserializer(IDataReader reader, Type type, IRecordStructure structure, SchemaMappingType mappingType) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\ClassDeserializerGenerator.cs:line 51
   at Insight.Database.CodeGenerator.DbReaderDeserializer.<>c__DisplayClass5.<GetDeserializer>b__4(SchemaMappingIdentity key) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\DbReaderDeserializer.cs:line 174
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Insight.Database.CodeGenerator.DbReaderDeserializer.GetDeserializer(IDataReader reader, Type type, IRecordStructure structure, SchemaMappingType mappingType) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\DbReaderDeserializer.cs:line 172
   at Insight.Database.CodeGenerator.DbReaderDeserializer.GetDeserializer[T](IDataReader reader, IRecordStructure structure) in c:\projects.net\Insight\Insight.Database\Insight.Database\CodeGenerator\DbReaderDeserializer.cs:line 104
   at Insight.Database.OneToOne`1.GetRecordReader(IDataReader reader) in c:\projects.net\Insight\Insight.Database\Insight.Database\Structure\OneToOne.cs:line 197
   at Insight.Database.DBReaderExtensions.<AsEnumerable>d__0`1.MoveNext() in c:\projects.net\Insight\Insight.Database\Insight.Database\Extensions\DbReaderExtensions.cs:line 75
   at Insight.Database.Structure.ListReader`1.Read(IDbCommand command, IDataReader reader) in c:\projects.net\Insight\Insight.Database\Insight.Database\Structure\ListReader.cs:line 55
   at Insight.Database.DBConnectionExtensions.<>c__DisplayClass69`1.<Query>b__68(IDbCommand cmd, IDataReader r) in c:\projects.net\Insight\Insight.Database\Insight.Database\Extensions\DBConnectionExtensions.cs:line 501
   at Insight.Database.DBConnectionExtensions.ExecuteAndAutoClose[T](IDbConnection connection, Func`2 getCommand, Func`3 translate, CommandBehavior commandBehavior) in c:\projects.net\Insight\Insight.Database\Insight.Database\Extensions\DBConnectionExtensions.cs:line 1461
   at Insight.Database.DBConnectionExtensions.Query[T](IDbConnection connection, String sql, Object parameters, IQueryReader`1 returns, CommandType commandType, CommandBehavior commandBehavior, Nullable`1 commandTimeout, IDbTransaction transaction, Object outputParameters) in c:\projects.net\Insight\Insight.Database\Insight.Database\Extensions\DBConnectionExtensions.cs:line 497
   at Insight.Database.DBConnectionExtensions.QuerySql[T1](IDbConnection connection, String sql, Object parameters, CommandBehavior commandBehavior, Nullable`1 commandTimeout, IDbTransaction transaction, Object outputParameters) in c:\projects.net\Insight\Insight.Database\Insight.Database\Extensions\DBConnectionExtensions.generated.cs:line 75
```

It's worth noting that returning null (whilst as indicated will allow other registered mappers to be 'tried'), functions correctly.

**Comment by kevinsea** [comment]
I've reviewed ColumnMapping.cs and I don't see implementation for '__unmapped__'.  I'd expect to see near line 146.  I think you are doing it correctly as it stands.

I'll leave this open until the documentation is revised or the feature is implemented.

**Comment by MatthewTrout** [comment]
Okay thanks for that. I'll take a look and see if I can do anything about it.

**Comment by jonwagner** [comment]
This is now fixed in 5.2.9.
